public/components/datasources/components/manage/associated_objects_tab.tsx

```diff
@@ -0,0 +1,176 @@
+  };
+
+  useEffect(() => {
+    fetchAssociatedObjects();
+  }, []);
+
+  const AssociatedObjectsHeader = () => {
+    return (
+      <EuiFlexGroup direction="row">
+        <EuiFlexItem>
+          <EuiText size="m">
+            <h2 className="panel-title">Associated objects</h2>
+            Manage objects associated with this data sources.
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiText>
+            <p>Last updated at: {lastUpdated}</p>
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            data-test-subj="freshButton"
+            iconType="refresh"
+            onClick={fetchAssociatedObjects}
+          >
+            Refresh
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  };
+
+  const columns = [
+    {
+      field: 'name',
+      name: 'Name',
+      sortable: true,
+      'data-test-subj': 'nameCell',
+      render: (name: string) => (
+        <EuiLink href="https://oui.opensearch.org/latest/" target="_blank">
+          {name}
+        </EuiLink>
+      ),
+    },
+    {
+      field: 'database',
+      name: 'Database',
+      truncateText: true,
+      render: (database: string) => (
+        <EuiLink href="https://oui.opensearch.org/latest/" target="_blank">
+          {database}
+        </EuiLink>
+      ),
+    },
+    {
+      field: 'type',
+      name: 'Type',
+      sortable: true,
+    },
+    {
+      field: 'createdByIntegration',
+      name: 'Created by Integration',
+      sortable: true,
+    },
+    {
+      field: 'accelerations',
+      name: 'Accelerations',
+      sortable: true,
+    },
+    {
+      name: 'Actions',
+      actions: [
+        {
+          name: 'Edit',
+          description: 'Edit this object',
+          type: 'icon',
+          icon: 'discoverApp',
+          onClick: (item: AssociatedObject) => console.log('Edit', item),
+        },
+        {
+          name: 'Delete',
+          description: 'Delete this object',
+          type: 'icon',
+          icon: 'bolt',
+          onClick: (item: AssociatedObject) => console.log('Delete', item),
+        },
+      ],
+    },
+  ];
+
+  const search = {
+    box: {
+      incremental: true,
+      schema: {
+        fields: { name: { type: 'string' }, database: { type: 'string' } }, // Adjust according to your data's fields
+      },
+    },
+  };
+
+  const pagination = {
+    initialPageSize: 10,
+    pageSizeOptions: [5, 10, 20],
+  };
+
+  const sorting = {
+    sort: {
+      field: 'name', // Default sort field
+      direction: 'asc', // Default sort direction
+    },
+  };
+
+  return (
+    <>
+      <EuiSpacer />
+      <EuiPanel>
+        <AssociatedObjectsHeader />
+        <EuiHorizontalRule />
+        <AccelerationsRecommendationCallout />
+        <EuiSpacer />
+        <EuiInMemoryTable
+          items={associatedObjects}
+          columns={columns}
+          search={search}
```

Comment:
May need to move this out for an external search bar component since the `EuiInMemoryTable` doesn't have support for search bar **and** filters, but that can be done in a future PR.

Reply:
lets do this as a follow up